### DISCUSSION TITLE
Login/ Register QoL improvement

### DISCUSSION THREAD

**Discussion by Tobiwue** [remsfal/remsfal-frontend#80] [Ideas]
Hey i have a quick idea about the login screen.
First I created an issue [here](https://github.com/remsfal/remsfal-frontend/issues/74) but Astanik asked me if I would like to start a discussion from it.
There may be other creative ideas that you can think of.
Let's come to my idea:
The current login screen sends you directly to a new page, although it would be better if a small login window was opened like in the example picture below.

![image](https://github.com/remsfal/remsfal-frontend/assets/76914772/4c705f91-a245-4ec2-8bfb-d611d282a91b)

It provides users with a seamless experience of the website.
If you want to continue browsing the main page before you decide to register, you don't necessarily have to reload the main page, you can simply close the popup.

**Comment by HassanAlhawari1998** [comment]
You can log in with Google Authentication using a pop-up window, without being redirected to another tab. This makes the process smoother for users.

To do this, use the URL like "auth/google" to open the pop-up. After logging in, the necessary information is sent back to the main window, where the login is completed.

You can find the complete code for this method here: https://dev.to/dinkydani21/how-we-use-a-popup-for-google-and-outlook-oauth-oci 

This way, users can log in securely without leaving the current tab.

**Reply by astanik** [reply]
Ok, it sounds very interesting. Do you want to try implementing this behavior?

**Reply by HassanAlhawari1998** [reply]
I have looked it up and I found that there could be potential risks in using Pop-Up's, when the implementation is not well handled.
Since I am unfortunately not so familiar with the implementation of such login popus, I would refrain from doing so.
